teachopencadd/talktorials/T018_automated_cadd_pipeline/utils/helpers/pdb.py

```diff
@@ -0,0 +1,184 @@
+    with open(full_filepath, "w") as f:
+        f.write(pdb_file_content)
+    return full_filepath
+
+
+def extract_molecule_from_pdb_file(molecule_name, input_filepath, output_filepath):
+    """
+    Extract a specific molecule (i.e. the protein or a ligand)
+    from a local PDB file and save as a new PDB file in a given path.
+
+    Parameters
+    ----------
+    molecule_name : str
+        Name of the molecule to be extracted.
+        For the protein, enter 'protein'. For a ligand, enter the ligand-ID.
+    input_filepath : str or pathlib.Path
+        Local path of the original PDB file.
+    output_filepath : str or pathlib.Path
+        Local file path (including file name) to save the PDB file of the extracted molecule in.
+
+    Returns
+    -------
+    MDAnalysis.core.universe.Universe
+        Structure object of the extracted molecule.
+    """
+
+    pdb_structure = Structure.from_string(input_filepath)
+    molecule_name = f"resname {molecule_name}" if molecule_name != "protein" else molecule_name
+    extracted_structure = pdb_structure.select_atoms(molecule_name)
+    extracted_structure.write(output_filepath)
+    return extracted_structure
+
+
+def load_pdb_file_as_dataframe(pdb_file_text_content):
+    """
+    Transform the textual content of a PDB file into a dictionary of pandas DataFrames.
+
+    Parameters
+    ----------
+    pdb_file_text_content : str
+        Textual content of a PDB file as a single string.
+
+    Returns
+    -------
+    Dict of pandas.DataFrames
+        The dictionary has 4 entries with following keys: 'ATOM', 'HETATM', 'ANISOU' and 'OTHERS'.
+        Each value is a DataFrame corresponding to the specific information described by the key.
+    """
+    ppdb = PandasPdb().read_pdb_from_list(pdb_file_text_content.splitlines(True))
+    pdb_df = ppdb.df
+    return pdb_df
+
+
+def extract_info_from_pdb_file_content(pdb_file_text_content):
+    """
+    Extract some useful information from the contents of a PDB file.
+
+    Parameters
+    ----------
+    pdb_file_text_content : str
+        Textual content of a PDB file as a single string.
+
+    Returns
+    -------
+    dict
+        Dictionary of the successfully extracted information.
+        Possible keys are:
+            'structure_title' : str
+                Title of the PDB structure.
+            'name' : str
+                Name of the protein.
+            'chains' : list of str
+                List of chain-IDs of the available chains in the protein.
+            'ligands' : list of list of str
+                List of ligand information:
+                [ligand-ID, chain-ID+residue number, number of heavy atoms]
+    """
+
+    pdb_content = pdb_file_text_content.strip().split("\n")
+
+    for index in range(len(pdb_content)):
+
+        # pdb_content[index]
+        # 'COMPND   4 FRAGMENT: KINASE DOMAIN, UNP RESIDUES 696-1022;                      '
+        pdb_content[index] = pdb_content[index].split(" ", 1)
```

Comment:
`.split()` default argument is already a space and tabs and other whitespace. If you want to add that default behaviour while specifying a single split, use `None` as the 1st arg.

Reply:
changed it to `.split(maxsplit=1)` for better readability (see 8cdfb36).